### PR TITLE
Kernels for max min mean median

### PR DESCRIPTION
**Pull request by Alexander-Makaryev**

### CODE REVIEW

File: dpnp/backend_statistics.pyx [hunk 1]
```diff
@@ -166,7 +189,26 @@ cpdef dparray dpnp_max(dparray input, axis):
     return dpnp_result_array
 
 
+cpdef dparray _dpnp_mean(dparray input):
+    cdef DPNPFuncType param1_type = dpnp_dtype_to_DPNPFuncType(input.dtype)
+    cdef DPNPFuncType output_type = dpnp_dtype_to_DPNPFuncType(numpy.float64 if input.dtype != numpy.float32 else numpy.float32)
```
Comment (densmirn) [inline dpnp/backend_statistics.pyx:194]: Isn't it resolved by `fmap[DPNPFuncName::DPNP_FN_MEAN][eft_FLT][eft_FLT] = {eft_FLT, (void*)custom_mean_c<float, float>};`?

File: dpnp/backend_statistics.pyx [hunk 2]
```diff
@@ -87,7 +91,26 @@ cpdef dparray dpnp_cov(dparray array1):
     return result
 
 
+cpdef dparray _dpnp_max(dparray input):
+    cdef DPNPFuncType param1_type = dpnp_dtype_to_DPNPFuncType(input.dtype)
+    cdef DPNPFuncType output_type = dpnp_dtype_to_DPNPFuncType(input.dtype)
```
[the remark this answers is not in the document]
Comment (densmirn) [reply dpnp/backend_statistics.pyx:96]: I thought that the third parameter of `get_dpnp_function_ptr` isn't output type, but it's type of the second input parameter. Please correct me if I'm wrong.

**Review by Rubtsowa — APPROVED** [review]

File: dpnp/backend/custom_kernels_statistics.cpp [hunk 3]
```diff
@@ -139,3 +139,146 @@ void custom_cov_c(void* array1_in, void* result1, size_t nrows, size_t ncols)
...
+
+    size_t size = 1;
+    for (size_t i = 0; i < ndim; ++i)
+    {
+        size *= shape[i];
+    }
+
+    _DataType* sum = reinterpret_cast<_DataType*>(dpnp_memory_alloc_c(1 * sizeof(_DataType)));
+
+    custom_sum_c<_DataType>(array1_in, sum, size);
```
Comment (shssf) [inline dpnp/backend/custom_kernels_statistics.cpp:191]:
```suggestion
    _DataType sum;

    custom_sum_c<_DataType>(array1_in, &sum, size);
```

Comment (shssf) [reply dpnp/backend/custom_kernels_statistics.cpp:191]: oh... no. we need USM memory for kernel. sorry - my proposal is wrong.
perhaps it is better to use `*result` memory for this

**Pull request merged**